src/internal/storage/fileset/index/reader.go

```diff
@@ -214,6 +232,7 @@ type ShardConfig struct {
 // A subtree is traversed only when a split point exists within it, which we know based on the NumFiles and SizeBytes
 // values at the root of each subtree.
 func (r *Reader) Shards(ctx context.Context) ([]*PathRange, error) {
+	ctx = ctxWithMetrics(ctx)
```

Comment:
I would remove this for now. Maybe `traversedRanges` and `traversedFiles` make sense, but I don't think this is as important as the iterate meters.

Reply:
I decided to keep this but to wrap another context around this so we can see its shards in the logs.